### PR TITLE
Cleanup and some plugin boilerplate for urls, data, and vfolders

### PR DESCRIPTION
**Pull request by phlax**
Adds

- ~~migration for vfolders to ensure that previously valid vfolders are renamed~~
- boilerplate delegate declarations for pootle_data and vfolders
- some useful fixtures from the test env

removing migration in favour of addressing - #5217

### CODE REVIEW

File: pootle/apps/virtualfolder/migrations/0004_cleanup_vfolder_names.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+def rename(vfolder):
+    i = 0
+    old_name = vfolder.name
+    while True:
```
Comment (dwaynebailey) [inline pootle/apps/virtualfolder/migrations/0004_cleanup_vfolder_names.py]: How does this ever exit?

Comment (phlax) [inline pootle/apps/virtualfolder/migrations/0004_cleanup_vfolder_names.py]: good point - i forgot the `break`

Comment (phlax) [inline pootle/apps/virtualfolder/migrations/0004_cleanup_vfolder_names.py]: i actually hate this pattern anyway - and its worth mentioning that this commit may die - in favour of addressing https://github.com/translate/pootle/issues/5217

File: pootle/apps/pootle_store/urls.py [hunk 2]
```diff
@@ -54,5 +51,11 @@
 
     url(r'^xhr/units/(?P<uid>[0-9]+)/checks/(?P<check_id>[0-9]+)/toggle/?$',
         views.toggle_qualitycheck,
-        name='pootle-xhr-units-checks-toggle'),
-]
+        name='pootle-xhr-units-checks-toggle')]
```
Comment (unho) [inline pootle/apps/pootle_store/urls.py]: This is just adding unneeded diff lines.

File: pytest_pootle/fixtures/models/translation_project.py [hunk 3]
```diff
@@ -92,6 +92,16 @@ def tp0(language0, project0):
 
 
 @pytest.fixture
+def tp1(language1, project1):
+    """Require English Project1."""
```
Comment (unho) [inline pytest_pootle/fixtures/models/translation_project.py]: Docstring needs to be changed.

**Comment by unho** [comment]
lgtm

**Pull request merged**